gammapy/detect/_test_statistics_cython.pyx

```diff
@@ -95,23 +99,27 @@ def _amplitude_bounds_cython(np.ndarray[np.float_t, ndim=2] counts,
     """
 
     cdef np.float_t s_model = 0, s_counts = 0, sn, sn_min = 1E14, c_min = 1
-    cdef np.float_t b_min, b_max
+    cdef np.float_t b_min, b_max, sn_min_total = 1E14
     cdef unsigned int i, j, ni, nj
     ni = counts.shape[1]
     nj = counts.shape[0]
     for j in range(ni):
         for i in range(nj):
-            s_model += model[j, i]
             if counts[j, i] > 0:
                 s_counts += counts[j, i]
                 if model[j, i] > 0:
                     sn = background[j, i] / model[j, i]
                     if sn < sn_min:
                         sn_min = sn
                         c_min = counts[j, i]
+            if model[j, i] > 0:
+                s_model += model[j, i]
+                sn = background[j, i] / model[j, i]
+                if sn < sn_min_total:
+                    sn_min_total = sn
     b_min = c_min / s_model - sn_min
     b_max = s_counts / s_model - sn_min
-    return b_min / FLUX_FACTOR, b_max / FLUX_FACTOR
+    return b_min / FLUX_FACTOR, b_max / FLUX_FACTOR, -sn_min_total / FLUX_FACTOR
```

Comment:
Can you change to return a `NamedTuple` or dict, please?

Calling code is always easier to read if functions that return multiple things give names to those things instead of using `result[0]`, `result[1]`, result[2]`.

There's a few more functions in Gammapy that return tuples, I suggest we change all of them to the more readable alternative.